compiler_gym/service/runtime/BenchmarkCache.h

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+#pragma once
+
+#include <grpcpp/grpcpp.h>
+
+#include <memory>
+#include <mutex>
+#include <optional>
+#include <random>
+
+#include "boost/filesystem.hpp"
+#include "compiler_gym/service/proto/compiler_gym_service.pb.h"
+
+namespace compiler_gym::runtime {
+
+constexpr size_t kEvictionSizeInBytes = 512 * 1024 * 1024;
+
+// An in-memory cache of Benchmark protocol buffers.
+//
+// This object caches Benchmark messages by URI. Once the cache reaches a
+// predetermined size, benchmarks are evicted randomly until the capacity is
+// reduced to 50%.
+class BenchmarkCache {
+ public:
+  BenchmarkCache(std::optional<std::mt19937_64> rand = std::nullopt,
+                 size_t maxSizeInBytes = kEvictionSizeInBytes);
```

Comment:
Totally not a thing, but should the order of these be reversed? Aren't they more likely to give a max size than a rand?

Reply:
Sure, done :)